### ISSUE TITLE
Error: Missing cursor in initial fetch response.

### ISSUE THREAD

**Issue by OlegShklyarov** [zerodytrash/TikTok-Live-Connector#47]
After the last update. What does that mean?

**Comment by lzyslh** [comment]
Same issue here.
This is the error log:

`Failed to connect Error: Missing cursor in initial fetch response.
    at WebcastPushConnection._fetchRoomData2 (/var/www/mydomain/files/node_modules/tiktok-live-connector/dist/index.js:503:11)
    at process._tickCallback (internal/process/next_tick.js:68:7)`

**Comment by OlegShklyarov** [comment]
They also banned my account where I took a session id from.
I have heard that it in not required anymore? Provided by the backend?

**Comment by addeeandra** [comment]
I just got the same issue :

`Connection error:  Missing cursor in initial fetch response.`

**Comment by OlegShklyarov** [comment]
Maybe Tt is changing something again. They banned my sessionid account.

**Comment by zerodytrash** [comment]
Yes, TikTok tries again to make the library unusable. Unbelievable...
I'm losing motivation for the project :(

**Comment by OlegShklyarov** [comment]
> Yes, TikTok tries again to make the library unusable. Unbelievable... I'm losing motivation for the project :(

What if we start a crowd funding campaign to keep you motivated?

**Comment by zerodytrash** [comment]
> What if we`ll start a crowd funding campaign to keep you motivated?

Sure, if I would get money for it I could invest more time in the project and try to keep it running as good as possible. So if anyone wants to support me, let me know.

**Comment by OlegShklyarov** [comment]
Create a Patreon, please. And Papal donate page.

**Comment by adierebel** [comment]
> Create a Patreon, please. And Paypal donate page.

Yes please

**Comment by OlegShklyarov** [comment]
Also add a funding message to the node update logs like I saw it on other libraries.

**Comment by zerodytrash** [comment]
Here is my paypal: https://www.paypal.com/paypalme/dalixz

After i know what TT changed i will also create a patreon page. Thanks for your support.

**Comment by OlegShklyarov** [comment]
> 
Good. We are waitong. I will promote your Paypal and Patreon across my clients.

**Comment by lzyslh** [comment]
> Yes, TikTok tries again to make the library unusable. Unbelievable... I'm losing motivation for the project :(

Why do you think so? For what it's worth, I think this library benefits TikTok the most, don't you agree?

**Comment by iProGamer** [comment]
I think anything that looks too good to be true, wont last, and this service is one of them

**Comment by OlegShklyarov** [comment]
> > Yes, TikTok tries again to make the library unusable. Unbelievable... I'm losing motivation for the project :(
> 
> Why do you think so? For what it's worth, I think this library benefits TikTok the most, don't you agree?

No it is not. Tiktok hates services likt this. Using them is against TOS. And he will always try to shut us down.

**Comment by affluentmind** [comment]
The interesting thing is , the python alternative is polling data from the webcast response but the node js isn't getting a response
on _fetchRoomData2 it just returns blank

**Comment by OlegShklyarov** [comment]
> 

Does Python version require sessionid and ttwid to connect?

**Comment by MrBill15** [comment]
Any fix for this problems?

**Comment by zerodytrash** [comment]
Have updated the signing server. It now works again for me. Can anyone confirm this?

**Comment by affluentmind** [comment]
so basically I made a workaround after the _fetchRoomData2

i added 

  do {
    
  } while (webcastResponse.cursor == '');


to force wait the first cursor

**Comment by iProGamer** [comment]
yes it work

**Comment by OlegShklyarov** [comment]
> Have updated the signing server. It now works again for me. Can anyone confirm this? (no npm update required)

It works for now, but we all know where it is all going... So please create a Patreon page. I will promote it starting tomorrow. All guys here, please support the author. He needs motivation to maintain this socket. Tiktok will hunt us harder and harder. Tough times ahead.

**Comment by lzyslh** [comment]
Working perfectly! Thank you, is there anything similar to this for Twitch?

**Comment by OlegShklyarov** [comment]
> Working perfectly! Thank you, is there anything similar to this for Twitch?

it is a well known library tmi.js, but I use Streamlabs socket.

**Comment by lzyslh** [comment]
> > Working perfectly! Thank you, is there anything similar to this for Twitch?
> 
> There is a well known library tmi.js, but I use Streamlabs socket.

Thank you, will check them out!

**Comment by MrBill15** [comment]
> Have updated the signing server. It now works again for me. Can anyone confirm this?
> (no npm update required)

It's mean we need to change the session id ?

**Comment by zerodytrash** [comment]
> It works for now, but we all know where it is all going... So please create a Patreon page.

@OlegShklyarov  https://www.patreon.com/zerody 😄

**Comment by tommy50773** [comment]
> > Have updated the signing server. It now works again for me. Can anyone confirm this? (no npm update required)
> 
> It works for now, but we all know where it is all going... So please create a Patreon page. I will promote it starting tomorrow. All guys here, please support the author. He needs motivation to maintain this socket. Tiktok will hunt us harder and harder. Tough times ahead.
> 
> Also add Patreon and Paypal links to your Heroku Demo.

After I generate the signature through the signature service and generate the url, it seems that I still cannot access it normally.

**Comment by OlegShklyarov** [comment]
> https://www.patreon.com/zerody

Thanks! Let see how the community supports you.

Is it just me or gift events come doubled? 

![image](https://user-images.githubusercontent.com/34983317/180987913-d2b86a6d-89f3-4249-8b4a-ce57b3ebf925.png)

**Comment by DevNetworktm** [comment]
@zerodytrash , hello , the problem has been fixed, it takes a while, but immediately, it no longer works :(

**Comment by DevNetworktm** [comment]
> Thanks! Let see how the community supports you.

> Is it just me or gift events come doubled?

@OlegShklyarov 

it is necessary to make a condition following the event function, it is indicated on the documentation

**Comment by renelima** [comment]
The error came back... Is it just for me?

**Comment by pbreit1988** [comment]
I have the same error

**Comment by MrBill15** [comment]
> Here is my paypal: https://www.paypal.com/paypalme/dalixz
> 
> After i know what TT changed i will also create a patreon page. Thanks for your support.

hi how about make discord too for this repo ? :D

**Comment by cchirila** [comment]
> The error came back... Is it just for me?

You are not alone, I have the same problem, it was working 1 hour ago, but now:

```
Failed to connect Error: Missing cursor in initial fetch response.
    at WebcastPushConnection._fetchRoomData2 (/home/admin/tiktok/node_modules/tiktok-live-connector/dist/index.js:503:11)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async WebcastPushConnection.connect (/home/admin/tiktok/node_modules/tiktok-live-connector/dist/index.js:268:7)
```

**Comment by affluentmind** [comment]
Signing server for node is returning blank again, /webcast/sign_url?client=ttlive-python&url= 
however returns a file that contains the next_cursor i'm not sure but maybe rewrite the signing process to parse next_cursor? 
i'm not sure how all of this works, just a thought

**Comment by lunat1k-12** [comment]
Same problem for me
`Error: Missing cursor in initial fetch response. at WebcastPushConnection._fetchRoomData2 (/Users/yvgenchemeris/workspace/tiktok_stream/node_modules/tiktok-live-connector/dist/index.js:503:11) at process.processTicksAndRejections (node:internal/process/task_queues:95:5) at async WebcastPushConnection.connect (/Users/yvgenchemeris/workspace/tiktok_stream/node_modules/tiktok-live-connector/dist/index.js:268:7) `

**Comment by OlegShklyarov** [comment]
It is sad that Tiktok provides no official api for live chat as it Twitch or Youtube do

**Comment by DevNetworktm** [comment]
> It is sad that Tiktok provides no official api for live chat as it Twitch or Youtube do

![image](https://user-images.githubusercontent.com/66486552/181075867-036e60aa-ab14-49c7-9e0c-aa48775862f2.png)

**Comment by guzinhoo** [comment]
I put the old version and used the session ID and it's working ok

**Comment by OlegShklyarov** [comment]
> I put the old version and used the session ID and it's working ok

Last time they banned my channel where I took sessionid from

**Comment by odhier** [comment]
> I put the old version and used the session ID and it's working ok

I've tried to use old version, it was connected but no events are triggered

**Comment by affluentmind** [comment]
> > I put the old version and used the session ID and it's working ok
> 
> I've tried to use old version, it was connected but no events are triggered

0.9.21 works
0.9.22 gets stuck after connecting

**Comment by OlegShklyarov** [comment]
> > > I put the old version and used the session ID and it's working ok
> > 
> > 
> > I've tried to use old version, it was connected but no events are triggered
> 
> 0.9.21 works 0.9.22 gets stuck after connecting

Works, but sessionid is required?

**Comment by guzinhoo** [comment]
> > I put the old version and used the session ID and it's working ok
> 
> I've tried to use old version, it was connected but no events are triggered

i had to delete tiktok live from node folder because it always downloads the newest version and i got a version 0.9.21, and i just used requestHeaders and put the sessionid without the ttwid

**Comment by OlegShklyarov** [comment]
How to install older version?

**Comment by guzinhoo** [comment]
> How to install older version?

I always save all versions in case there is a problem with the current one I have the old one.

**Comment by affluentmind** [comment]
> How to install older version?

use  npm install tiktok-live-connector@version
version being the version number you want to install

**Comment by Daviko290** [comment]
> ¿Cómo instalar una versión anterior?

i did it this way.  npm i tiktok-live-connector@0.9.21
then with npm i list you verify that it is in version 0.9.21

**Comment by OlegShklyarov** [comment]
I will try, thanks

**Comment by dwins-dev** [comment]
> > How to install older version?
> 
> i did it this way. npm i tiktok-live-connector@0.9.21 then with npm list you verify that it is in version 0.9.21

Installed version 0.9.21
At the same time, I used a proxy and sessionId
It helped to connect without errors and for some time to receive events from the air
But after a few minutes, the events stop coming.
Perhaps I'm missing something?

**Comment by OlegShklyarov** [comment]
> > > How to install older version?
> > 
> > 
> > i did it this way. npm i tiktok-live-connector@0.9.21 then with npm list you verify that it is in version 0.9.21
> 
> Version installed 0.9.21 In doing so, I used a proxy and sessionId on the client side Helped to connect without errors and receive events from the air for some time But after a few minutes the events stop. Perhaps I'm missing something?

Yes, it just stops after some time.

**Comment by MrBill15** [comment]
Any hint to fix this issues?

**Comment by OlegShklyarov** [comment]
They are looking for a solution. Remember you can support the author and keep them active:
https://www.paypal.com/paypalme/dalixz
https://www.patreon.com/zerody

**Comment by Daviko290** [comment]
today everything works fine

**Comment by OlegShklyarov** [comment]
> 

Last version without Sessionid?

**Comment by iProGamer** [comment]
> > 
> 
> Last version without Sessionid?

yes

**Comment by OlegShklyarov** [comment]
Hoping it will last longer.

**Comment by OlegShklyarov** [comment]
connection2.js:30 Error: Failed to sign request: Request failed with status code 429; URL: https://webcast.tiktok.com/webcast/im/fetch/?aid=1988&app_language=en-US&app_name=tiktok_web&browser_language=en&browser_name=Mozilla&browser_online=true&browser_platform=Win32&browser_version=5.0+%28Windows+NT+10.0%3B+Win64%3B+x64%29+AppleWebKit%2F537.36+%28KHTML%2C+like+Gecko%29+Chrome%2F102.0.5005.63+Safari%2F537.36&cookie_enabled=true&cursor=&internal_ext=&device_platform=web&focus_state=true&from_page=user&history_len=4&is_fullscreen=false&is_page_visible=true&did_rule=3&fetch_rule=1&identity=audience&last_rtt=0&live_id=12&resp_content_type=protobuf&screen_height=1152&screen_width=2048&tz_name=Europe%2FBerlin&referer=https%3A%2F%2Fwww.tiktok.com%2F&root_referer=https%3A%2F%2Fwww.tiktok.com%2F&version_code=180800&webcast_sdk_version=1.3.0&update_version_code=1.3.0&room_id=7125065513440774913

**Comment by OlegShklyarov** [comment]
What is their request quota?

**Comment by Almuftres** [comment]
So can anyone know how the tiktok user: guess_the_word_game

can still stream his game ?

**Comment by OlegShklyarov** [comment]
> So can anyone know how the tiktok user: guess_the_word_game
> 
> can still stream his game ?

Maybe he is using a custom session id and dedicated IP.

**Comment by acaksandi** [comment]
its happen again "Failed to connect Error: Missing cursor in initial fetch response."

**Comment by xlzengin** [comment]
its happen again "Failed to connect Error: Missing cursor in initial fetch response."

**Comment by TeddyInWeb** [comment]
its happen again "Failed to connect Error: Missing cursor in initial fetch response."

**Comment by zerodytrash** [comment]
its happen again "Failed to connect Error: Missing cursor in initial fetch response."

**Comment by affluentmind** [comment]
Just set a sessionId in your code and use request polling by disabling the signing on the index.js file for the time being

**Comment by Daviko290** [comment]
it works again

**Comment by Daviko290** [comment]
> Just set a sessionId in your code and use request polling by disabling the signing on the index.js file for the time being

How could I disable that in the index.js

**Comment by tituo2021** [comment]
it seems that the python counterpart has solved. So we should donate more and the author would more enthusiasm on the project.

**Comment by isaackogan** [comment]
> it seems that the python counterpart has solved. So we should donate more and the author would more enthusiasm on the project.

wow ok not to be that guy but no one donates to me like that D=
Jokes aside, his library works. Tested just now.

![img](https://cdn.isaackogan.com/zsV9m)

**Comment by isaackogan** [comment]
> Just set a sessionId in your code and use request polling by disabling the signing on the index.js file for the time being

No need, fixed

**Comment by affluentmind** [comment]
> > Just set a sessionId in your code and use request polling by disabling the signing on the index.js file for the time being
> 
> No need, fixed

Confirmed, works again.
It's your signing server, or at least I assume so, could we at least know what was the problem? why was the signing a problem?

**Comment by Daviko290** [comment]
> > it seems that the python counterpart has solved. So we should donate more and the author would more enthusiasm on the project.
> 
> wow ok not to be that guy but no one donates to _me_ like that D:
> 
> Jokes aside, his library works. Tested just now. If it is not working for you, run `npm update`.
> 
> ![img](https://camo.githubusercontent.com/911ed658ce2023de36e2fd8dc057923f9c7819f76c9eea9f86b4274d074d5ece/68747470733a2f2f63646e2e69736161636b6f67616e2e636f6d2f7a7356396d)

working thanks

**Comment by isaackogan** [comment]
> It's your signing server, or at least I assume so, could we at least know what was the problem? why was the signing a problem?

what gave it away 😂

Sorry to be purposefully vague, but TikTok updated their WebCast servers to detect that our connection to their servers was suspcious. We had to find a way to get around that and stop being detected. We actually knew for a little while a block was coming, the new version was just not uploaded until it did.

**Comment by affluentmind** [comment]
> > It's your signing server, or at least I assume so, could we at least know what was the problem? why was the signing a problem?
> 
> what gave it away 😂
> 
> Sorry to be purposefully vague, but TikTok updated their WebCast servers to detect our signature generation. We had to find a way to get around that and mitigate detection. We actually knew for a little while a block was coming, we just waited until it broke until we uploaded the new version.

I understand, thanks for your answer.
To be honest, the domain name did  😂😂😂

**Comment by tituo2021** [comment]
> > it seems that the python counterpart has solved. So we should donate more and the author would more enthusiasm on the project.
> 
> wow ok not to be that guy but no one donates to _me_ like that D:
> 
> Jokes aside, his library works. Tested just now. If it is not working for you, run `npm update`.
> 
> ![img](https://camo.githubusercontent.com/911ed658ce2023de36e2fd8dc057923f9c7819f76c9eea9f86b4274d074d5ece/68747470733a2f2f63646e2e69736161636b6f67616e2e636f6d2f7a7356396d)

Did a little bit. Hope you not disappoited. Tons of appreciations.

**Comment by isaackogan** [comment]
> Did a little bit with PP. Hope you not disappoited. Tons of appreciations. Danke schön!

I was only joking. Thank you so much though, I'll go get a coffee and keep working :)

**Comment by kadinjerry** [comment]
Why still can't connect：
![image](https://user-images.githubusercontent.com/89059343/181904850-408e1add-a19d-42c2-9c5b-2272d46638e8.png)

**Comment by AnasQiblawi** [comment]
For some reason, this actually fixed the problem.
```
npm update
```

**Comment by dwins-dev** [comment]
> По какой-то причине это фактически решило проблему.
> 
> ```
> npm update
> ```

In my case, this solves the problem for the first 2 connections to different accounts, after which it stops working again

**Comment by isaackogan** [comment]
> AnasQiblawi

The package received an update. Updating the package... fixes issues in the outdated version.

**Comment by isaackogan** [comment]
> In my case, this solves the problem for the first 2 connections to different accounts, after which it stops working again

Rate limiting 👍

**Comment by dwins-dev** [comment]
> Rate limiting 👍
Is it possible to bypass this?
I tried both through a proxy and without it, the result is the same, it starts to give this error
Please tell me which direction should I go?

**Comment by zerodytrash** [comment]
> Is it possible to bypass this?

We do not support bypassing the implemented rate limits. If the library is used again to scrape massive amounts of data then TikTok will take further action. Therefore, this behavior is good.

**Comment by snappy1024** [comment]
I've had this problem for a while.
I also tried many ways and finally found a possible result.
Seems to have different results in different regions.
When I connect the VPN to Japan, South Korea, Taiwan it results in "Missing cursor in initial fetch response", 
but when I connect the VPN to other countries it works fine. (Same code, same live room, same time.)
Are server verifications different in different regions?
Any suggestions how I can get it to work without a VPN?

**Comment by asdfg314284230** [comment]
I think I have the same question again, I can launch paypal and I'm going to sponsor it to provide the impetus

**Comment by asdfg314284230** [comment]
Or is there a payment method that supports mainland China?  For example, Alipay @zerodytrash

**Comment by asdfg314284230** [comment]
This problem occurs when multiple users are connected at the same time, but not when three users are connected.

**Comment by asdfg314284230** [comment]
But I was able to use Tiktok in the last two newly updated versions of the app, though some of its functions didn't work properly.

**Comment by asdfg314284230** [comment]
Because I need this project as a server to connect and deliver bullet screen data to multiple clients, it will be troublesome if there is the above rate limit problem

**Comment by asdfg314284230** [comment]
I want to be your first King contributor, but I don't have a proper payment method!  We really hope you can update this project

**Comment by isaackogan** [comment]
> Because I need this project as a server to connect and deliver bullet screen data to multiple clients, it will be troublesome if there is the above rate limit problem

If you need to use this commercially, contact via `fallen#9745` and we can set something up with Zerody. The rate limit is intentional, TikTok block this library when people scrape large amounts of data. 

The compromise is to make sure that clients of the library are respecting fair use and not, say, making 30 connections at once. If that is something you need for your use case, please send me a DM on Discord. It will cost money.

Isaac

**Comment by asdfg314284230** [comment]
> > 因为我需要这个项目作为服务器连接和下发弹幕数据给多个客户端，如果出现上述限速问题就麻烦了
> 
> 如果您需要在商业上使用它，请联系 via `fallen#9745`，我们可以使用 Zerody 进行设置。速率限制是故意的，当人们抓取大量数据时，TikTok 会阻止这个库。
> 
> 折衷方案是确保图书馆的客户尊重合理使用，而不是一次建立 30 个连接。如果这是您的用例所需要的，请在 Discord 上给我发送 DM，我们可以讨论托管服务来处理您正在寻找的内容。
> 
> tl;dr 这些库是用于黑客的，而不是公司。
> 
> 艾萨克

Okay, I think I see what you mean.  I'll keep in touch with you by email. I'm sorry that I'm not good at using Discord. I'll keep in touch with you by email.

**Comment by ThanoFish** [comment]
I'm still getting this error. Is there a known fix?

**Comment by isaackogan** [comment]
> I'm still getting this error. Is there a known fix?

Update...

**Comment by ThanoFish** [comment]
> > I'm still getting this error. Is there a known fix?
> 
> Update...

I did though... 

Now I notice that it's actually only for one person. I try connecting to 3 people, and only 1 of them gives this error. The rest say "LIVE has ended"

**Comment by isaackogan** [comment]
There is a limit for how many people you can connect to using this library at once

**Comment by affluentmind** [comment]
Failed to connect Error: Failed to sign request: Request failed with status code 500;

Connecting through a SessionId throws a missing cursor error

**Comment by xlzengin** [comment]
> > > I'm still getting this error. Is there a known fix?
> > 
> > 
> > Update...
> 
> I did though...
> 
> Now I notice that it's actually only for one person. I try connecting to 3 people, and only 1 of them gives this error. The rest say "LIVE has ended"
if you use for one person,but  when it runs for a few minutes,it will hapan again "missing cursor..."

**Comment by ThanoFish** [comment]
This really sucks that you can't connect to more than 2 people.
I just need to make an app which just notifies you when someone out of 4 people goes live, and you can't really do that as you can only have 2 WebcastPushConnection variables, declaring any more will just make everything say "Missing cursor ...."
Connecting to multiple people at the same time seems like basic functionality and it's annoying when you can't even do that...

I don't get why TikTok doesn't just make a public API for livestreams. If you want to stop these libraries, why not make a replacement?

**Comment by isaackogan** [comment]
Hi, yeah, such is life. 

It's either two, or zero. If unlimited access is easily available, TikTok will have to put developers towards blocking our library. This is a happy medium that keeps them away. 

There have already been several attempts to use this library to spam TikTok, and they justly responded by blocking it.

There are not many non-commercial use cases where people can justify connecting to multiple clients. 

If you need more, you can talk to myself or Zerody about acquiring increased limits.

**Comment by ThanoFish** [comment]
> It's either two, or zero. If unlimited access is easily available, TikTok will have to put developers towards blocking our libraries. This is a happy medium that keeps them away.

Is the issue with allowing to connect to multiple people, or with spamming requests? If it's about spamming, can there be something like for example 1 minute between each connect() call or something like that? I feel like that would be a lot less limiting.

**Comment by adierebel** [comment]
Personally, i love how @isaackogan limit their signing server. Maybe tiktok noticed since this kind of library became popular and many users used it. As the creator said, this libs is for hack only, if any people need more freedom you can contact @isaackogan to increase limit (i hope this not free, since they need money to paid the server bill & coffee)

**Comment by isaackogan** [comment]
> Personally, i love how @isaackogan limit their signing server. Maybe tiktok noticed since this kind of library became popular and many users used it. As the creator said, this libs is for hack only, if any people need more freedom you can contact @isaackogan to increase limit (i hope this not free, since they need money to paid the server bill & coffee)

Yeah, they were viewing the repository and a day later it got blocked.

**Comment by ThanoFish** [comment]
> > It's either two, or zero. If unlimited access is easily available, TikTok will have to put developers towards blocking our libraries. This is a happy medium that keeps them away.
> 
> Is the issue with allowing to connect to multiple people, or with spamming requests? If it's about spamming, can there be something like for example 1 minute between each connect() call or something like that? I feel like that would be a lot less limiting.

**Comment by zerodytrash** [comment]
We have an eye on the utilization of the signing server and can adjust the limits if there is less spam.
The general problem behind the limitations is that TikTok can detect and block our signatures if a lot of requests are traced back to our signature server. Our goal is to allow normal users to continue using the libraries. Power users who establish multiple connections to different streams simultaneously are difficult to distinguish from scrapers. Therefore, the whole thing is not so easy to handle.
 
> Is the issue with allowing to connect to multiple people, or with spamming requests? If it's about spamming, can there be something like for example 1 minute between each connect() call or something like that? I feel like that would be a lot less limiting.

Assume a data scraper has 20 proxies. This means 20 connections per minute. It would allow the data scraper to establish 1200 connections in one hour. The connections would remain until the live stream is terminated. Because we know that people do this crap, we have taken further measures that unfortunately also restrict power users.

However, there is still the option to obtain an API key to sign more requests.

> If you need to use this commercially, contact via fallen#9745 and we can set something up with Zerody.

**Comment by reverse4free** [comment]
> 

Maybe you can test this project who provides the signature stably and efficiently. https://github.com/reverse4free/TiktokPerseusWeb

**Comment by nikolay-esipov** [comment]
Hello, after updating to 1.0.1, everything was fine at first, but after a day, an error occurs: Error: Missing cursor in initial fetch response. I use from one account constantly. How can this be fixed?

**Comment by iProGamer** [comment]
same

**Comment by isaackogan** [comment]
> Maybe you can test this project who provides the signature stably and efficiently. 
*paid service

**Comment by crypxeo** [comment]
Unfortunately the same error occurs from yesterday :(((
**Error: Missing cursor in initial fetch response.**
Anyone know how to fix it?

**Comment by Norbiros** [comment]
Same here.

**Comment by AlanWang88** [comment]
Happy news, after a while of work, we've cracked Tiktok's latest signature algorithm, you can use our endpoint at [here](https://rapidapi.com/AlanWang/api/tiktokperseus2)

**Comment by isaackogan** [comment]
> Happy news, after a while of work, we've cracked Tiktok's latest signature algorithm, you can use our endpoint at [here](https://rapidapi.com/AlanWang/api/tiktokperseus2)

Paid service*

**Comment by AlanWang88** [comment]
> > Happy news, after a while of work, we've cracked Tiktok's latest signature algorithm, you can use our endpoint at [here](https://rapidapi.com/AlanWang/api/tiktokperseus2)
> 
> Paid service*

Yeah, to keep us active, TikTok always update there strategy, It is also to ensure that it is not abused, We also provide a free requests

**Comment by xmanandjessie** [comment]
I can't use it. How can I pay for the services you can use?

**Comment by DrBlender** [comment]
Today is the day. it happend again

Failed to connect Error: Missing cursor in initial fetch response.

**Comment by LegendNed** [comment]
> Today is the day. it happend again
> 
> Failed to connect Error: Missing cursor in initial fetch response. Connected to Unity WebSocket Server Failed to connect Error: Missing cursor in initial fetch response. at WebcastPushConnection._fetchRoomData2 (C:\TikTok\node_modules\tiktok-live-connector\dist\index.js:555:13) at process.processTicksAndRejections (node:internal/process/task_queues:95:5) at async WebcastPushConnection.connect (C:\TikTok\node_modules\tiktok-live-connector\dist\index.js:282:7)

They update the api soo much, i am no expert how to fix it but i did some debugging in hopes developers fix it quicker 😦

API url has not changed, the way they send the data has changed, they used to send response data but now that is at the most bottom request, in im my case being WebcasChatMessage, wsUrl changed to imprp , cursor changed to next_cursor . Hopefully i have the correct information, as there is no mentions of cursors now in the request.

**Comment by Williamalong23** [comment]
Quickly massage 22 cracker on Instagram for quick recovery now https://instagram.com/22cracker?igshid=YWYwM2I1ZDdmOQ==

**Comment by rna98mod** [comment]
any solution to this error ? this error occurs again just today

**Comment by YttriumNode** [comment]
It is already fixed!
Thanks, @zerodytrash

**Comment by isaackogan** [comment]
Fixed by Zerody

**Comment by regodo** [comment]
Hi all!

> I am developing an app which collects statistical data of TikTok lives.
> I use proxies for my connections to be able to make multiple connections safely.

I have a problem:

**Is it possible the I got this error even if I call the more "new WebcastPushConnection" more then 4 times?**
```javascript
this.tiktokLiveConnection =  new WebcastPushConnection( user1, { enableExtendedGiftInfo: true });
this.tiktokLiveConnection =  new WebcastPushConnection( user2, { enableExtendedGiftInfo: true });
this.tiktokLiveConnection =  new WebcastPushConnection( user3, { enableExtendedGiftInfo: true });
this.tiktokLiveConnection =  new WebcastPushConnection( user4, { enableExtendedGiftInfo: true });
this.tiktokLiveConnection =  new WebcastPushConnection( user5, { enableExtendedGiftInfo: true });
``` 
Even I can't use the same instance but with a new user (e.g: if the past user is offline)

After I initialized the 5 instance all of them get this error : "[Error: Missing cursor in initial fetch response.](https://github.com/zerodytrash/TikTok-Live-Connector/issues/47#top)"

Is there any limit of the plugin? Or is there anything else outside of the plugin which limits the plugin instances?
> this is the same with proxies or without proxies. Apparently for only 4 connections no need to use any proxies, according to my experience

I successfully bypass the problem by run the code multiply times from console; simple run the code twice. So I could use even 12 (3x4) instances.
> Notice: In this cases there were some instances that were just created and there was not connected for each (becasue of some of the was offline)

Even does my problem connected to this error? 

If someone can help me please message me!

**Comment by esalvi** [comment]
It happens to me again. Version: 1.1.5

**Comment by Eugeny335** [comment]
> It's happening to me again. Version: 1.1.5

Did you fix it?It started for me too

**Comment by esalvi** [comment]
No, I didn't. It worked yesterday (more or less...) but not today :-(

**Comment by DRIVER1ksa** [comment]
today not working

**Comment by fangzeyu1** [comment]
not working now . 2024-04-07
